### PR TITLE
Enable some IDNA tests on Unix

### PR DESCRIPTION
**Pull request by eerhardt**
1. Some IdnMapping tests were still disabled against #810.  Enabling these tests.

2. Fixing a small error on IdnMapping on Unix which was not failing with embedded null characters and other illegal characters.

Partial fix for #3406.

@ellismg @tarekgh @stephentoub

### CODE REVIEW

File: src/Common/src/System/Globalization/IdnMapping.Unix.cs [hunk 1]
```diff
@@ -67,5 +73,30 @@ private uint Flags
...
+        /// is enabled.
+        ///
+        /// To match Windows behavior, we walk the string ourselves looking for these
+        /// bad characters so we can continue to throw ArgumentException in these cases. 
+        /// </summary>
+        private static void CheckInvalidIdnCharacters(string s, uint flags, string paramName)
+        {
+            if ((flags & Interop.GlobalizationNative.UseStd3AsciiRules) == 0)
+            {
+                for (int i = 0; i < s.Length; i++)
```
Comment (tarekgh) [inline src/Common/src/System/Globalization/IdnMapping.Unix.cs]: this looks ok to me. if we think want to optimize the loop we can do something like the following.

            int length = s.Length;
            int index = 0;
            bool result = false;

            while (length > 0)
            {
                int len = Math.Min(length, 10);
                switch (len)
                {
                    case 10: c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 9:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 8:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 7:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 6:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 5:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 4:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 3:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 2:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                    case 1:  c = s[index]; result |= (c <= 0x1F || c == 0x7F); index++;
                }

                if (result)
                {
                    throw new ArgumentException(SR.Argument_IdnIllegalName, paramName); 
                }

                length -= len;
            }

**Comment by tarekgh** [comment]
LGTM.

**Comment by ellismg** [comment]
LGTM.  Thanks, Eric!

File: src/Common/src/System/Globalization/IdnMapping.Unix.cs (hunk 1 above)
Comment (eerhardt) [inline src/Common/src/System/Globalization/IdnMapping.Unix.cs]: I'm not sure I follow how this code works. Does it check every character?

Comment (stephentoub) [inline src/Common/src/System/Globalization/IdnMapping.Unix.cs]: It's basically a manual loop unrolling, but I suggest we not complicate the code as such unless we find there's actually a perf problem.

**Comment by stephentoub** [comment]
LGTM

File: src/Common/src/System/Globalization/IdnMapping.Unix.cs (hunk 1 above)
Comment (tarekgh) [inline src/Common/src/System/Globalization/IdnMapping.Unix.cs]: @stephentoub I agree. we just want to watch that.

Comment (JonHanna) [inline src/Common/src/System/Globalization/IdnMapping.Unix.cs]: I'd look at changing the check from `(c <= 0x1F || c == 0x7F)` to `(c <= 0x1F | c == 0x7F)` and removing the branch that only short-circuits on an equality-check (and that will be hit in the more common case anyway) before looking at unrolling.

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please
Test Innerloop Windows_NT Debug Build and Test please
Test Innerloop CentOS7.1 Release Build and Test please
Test Innerloop CentOS7.1 Debug Build and Test please

**Pull request merged**